### PR TITLE
Rstudio desktop2

### PR DESCRIPTION
**Pull request by izahn**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by isuruf** [comment]
This seems to fail with
```
CMake Error at src/cpp/desktop/CMakeLists.txt:96 (message):
  Did not find supported Qt SDK
```

**Comment by izahn** [comment]
> This seems to fail with
> 
> ```
> CMake Error at src/cpp/desktop/CMakeLists.txt:96 (message):
>   Did not find supported Qt SDK
> ```

Yeah sorry, that's just a bug and unrelated to the real issues I had with trying to get a freedesktop-like layout. Fixed now. 

As you can see the package builds, but when I run it I get this:
```
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/home/izahn/.config/kdedefaults:/etc/xdg/rstudio/logging.conf'. Logging to '/home/izahn/.local/share/rstudio/log/rdesktop.log'.
2022-03-19T14:53:08.987141Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/error.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::SessionLauncher::showLaunchErrorPage() src/cpp/desktop/DesktopSessionLauncher.cpp:379
"Uncaught TypeError: Cannot read property 'notifyRCrashed' of undefined"
2022-03-19T14:53:09.051025Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:09.069802Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:09.304886Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:09.724277Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:10.344989Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:11.163897Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:12.183333Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:13.403856Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T14:53:14.827088Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
```

I uploaded it to my `izahn` channel to make it easier to install and test locally if you want.

**Comment by isuruf** [comment]
It looks like we need to pass `-DRSTUDIO_PACKAGE_BUILD=1` to cmake regardless of this PR or the previous PR.
Also we need to patch https://github.com/rstudio/rstudio/blob/edd144e6c64f765f62716e57a31808597e92ab7e/src/cpp/desktop/DesktopOptions.cpp#L456 to be
```
return supportingFilePath_.completePath("share/rstudio");
```

**Comment by izahn** [comment]
> It looks like we need to pass `-DRSTUDIO_PACKAGE_BUILD=1` to cmake regardless of this PR or the previous PR. 

This causes it to build its own `libqminimal.so` instead of using the one in the `qt` package, I don't think we want to set this variable.

**Comment by izahn** [comment]
>Also we need to patch https://github.com/rstudio/rstudio/blob/edd144e6c64f765f62716e57a31808597e92ab7e/src/cpp/desktop/DesktopOptions.cpp#L456 to be
> 
> ```
> return supportingFilePath_.completePath("share/rstudio");
> ```

I patched as you suggested, but running the resulting binary still does not work and produces
```
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/home/izahn/.config/kdedefaults:/etc/xdg/rstudio/logging.conf'. Logging to '/home/izahn/.local/share/rstudio/log/rdesktop.log'.
2022-03-19T22:39:59.834043Z [rdesktop] WARNING r-ldpath script not found at /home/izahn/mambaforge/envs/rstudio/share/rstudio/session/r-ldpath; LOGGED FROM: std::string rstudio::core::r_util::{anonymous}::extraLibraryPaths(const rstudio::core::FilePath&, const string&) src/cpp/core/r_util/REnvironmentPosix.cpp:83
2022-03-19T22:40:00.529932Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/error.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::SessionLauncher::showLaunchErrorPage() src/cpp/desktop/DesktopSessionLauncher.cpp:379
"Uncaught TypeError: Cannot read property 'notifyRCrashed' of undefined"
2022-03-19T22:40:00.558093Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T22:40:00.571778Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T22:40:00.790805Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T22:40:01.213156Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T22:40:01.833111Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T22:40:02.650621Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T22:40:03.668797Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
2022-03-19T22:40:04.887589Z [rdesktop] ERROR system error 2 (No such file or directory) [path: /home/izahn/mambaforge/envs/rstudio/bin/resources/html/loading.html]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: void rstudio::desktop::MainWindow::onLoadFinished(bool) src/cpp/desktop/DesktopMainWindow.cpp:714
```
I feel like we're making this hard on ourselves, and I don't really see the benefit. https://github.com/conda-forge/staged-recipes/pull/18320 works (for sure on linux, maybe on osx too, I just don't have osx available to test). Why don't we just use that?

**Comment by isuruf** [comment]
Since `RSTUDIO_PACKAGE_BUILD` is not set, we have to patch https://github.com/rstudio/rstudio/blob/edd144e6c64f765f62716e57a31808597e92ab7e/src/cpp/desktop/DesktopOptions.cpp#L463 to be 1.

**Comment by izahn** [comment]
I've patched the sources to install to a standard directory layout, but running it produces these errors:
```
2022-03-23T11:32:32.794283Z [rsession-conda] ERROR system error 2 (No such file or directory) [path: $PREFIX/resources/schema/user-prefs-schema.json]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: rstudio::core::Error rstudio::session::prefs::Preferences::readLayers() src/cpp/session/prefs/Preferences.cpp:64
2022-03-23T11:32:32.795086Z [rsession-conda] ERROR system error 2 (No such file or directory) [path: $PREFIX/resources/schema/user-state-schema.json]; OCCURRED AT rstudio::core::Error rstudio::core::FilePath::openForRead(std::shared_ptr<std::basic_istream<char> >&) const src/cpp/shared_core/FilePath.cpp:1461; LOGGED FROM: rstudio::core::Error rstudio::session::prefs::Preferences::readLayers() src/cpp/session/prefs/Preferences.cpp:64
2022-03-23T11:32:32.795462Z [rsession-conda] WARNING Attempt to look up preference 'restore_last_project' before preference layer 'default' was read; LOGGED FROM: boost::optional<T> rstudio::session::prefs::PrefLayer::readPref(const string&) [with T = bool  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:56
2022-03-23T11:32:32.795579Z [rsession-conda] ERROR jsonrpc error 8 (Parameter missing) [description: missing default value for preference 'restore_last_project']; OCCURRED AT T rstudio::session::prefs::Preferences::readPref(const string&) [with T = bool  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:68; LOGGED FROM: T rstudio::session::prefs::Preferences::readPref(const string&) [with T = bool  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:71
2022-03-23T11:32:32.797919Z [rsession-conda] WARNING Attempt to look up preference 'initial_working_directory' before preference layer 'default' was read; LOGGED FROM: boost::optional<T> rstudio::session::prefs::PrefLayer::readPref(const string&) [with T = std::__cxx11::basic_string<char>  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:56
2022-03-23T11:32:32.798070Z [rsession-conda] ERROR jsonrpc error 8 (Parameter missing) [description: missing default value for preference 'initial_working_directory']; OCCURRED AT T rstudio::session::prefs::Preferences::readPref(const string&) [with T = std::__cxx11::basic_string<char>  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:68; LOGGED FROM: T rstudio::session::prefs::Preferences::readPref(const string&) [with T = std::__cxx11::basic_string<char>  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:71
2022-03-23T11:32:32.801697Z [rsession-conda] WARNING Attempt to look up preference 'ansi_console_mode' before preference layer 'default' was read; LOGGED FROM: boost::optional<T> rstudio::session::prefs::PrefLayer::readPref(const string&) [with T = std::__cxx11::basic_string<char>  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:56
2022-03-23T11:32:32.801849Z [rsession-conda] ERROR jsonrpc error 8 (Parameter missing) [description: missing default value for preference 'ansi_console_mode']; OCCURRED AT T rstudio::session::prefs::Preferences::readPref(const string&) [with T = std::__cxx11::basic_string<char>  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:68; LOGGED FROM: T rstudio::session::prefs::Preferences::readPref(const string&) [with T = std::__cxx11::basic_string<char>  std::string = std::__cxx11::basic_string<char>] src/cpp/session/prefs/UserPrefValues.cpp:71
2022-03-23T11:32:32.803405Z [rsession-conda] WARNING ASSERTION FAILED: px != 0; LOGGED FROM: typename boost::detail::sp_member_access<T>::type boost::shared_ptr<T>::operator->() const [with T = rstudio::core::json::Object  typename boost::detail::sp_member_access<T>::type = rstudio::core::json::Object*] $PREFIX/include/boost/smart_ptr/shared_ptr.hpp:728
```
Any other suggestions? I'm still inclined to abandon this effort, but willing to continue a bit longer if people think its important.

File: recipes/rstudio/0003-standard-directory-layout.patch [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+diff --git a/src/cpp/session/session-options.json b/src/cpp/session/session-options.json
+index c361250679..db0d51f4a5 100644
+--- a/src/cpp/session/session-options.json
++++ b/src/cpp/session/session-options.json
+@@ -560,7 +560,7 @@
+             "name": "r-resources-path",
+             "type": "core::FilePath",
+             "memberName": "rResourcesPath_",
+-            "defaultValue": "resources",
++            "defaultValue": "share/resources",
```
Comment (isuruf) [inline recipes/rstudio/0003-standard-directory-layout.patch:75]:
```suggestion
+            "defaultValue": "share/rstudio/resources",
```

**Comment by isuruf** [comment]
I think we need to fix https://github.com/rstudio/rstudio/blob/dad40bb763a777cffccc321de48d2d6e5dc39fa1/src/cpp/session/include/session/SessionOptions.gen.hpp#L304 too

**Comment by izahn** [comment]
All the builds here that have attempted to patch the source to install to a standard directory layout have failed to produce working binaries. Even if we manage to do that we may still have produced builds that run but are broken in less obvious ways. I am unwilling to accept responsibility for that, and so am closing this PR. I will leave my fork up in case someone else wants to pick this up later.

**Pull request closed without merging**